mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterAkkaImpl.java

```diff
@@ -0,0 +1,168 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.master.resourcecluster;
+
+import akka.actor.ActorRef;
+import akka.pattern.Patterns;
+import io.mantisrx.common.Ack;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.GetAvailableTaskExecutorsRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.GetBusyTaskExecutorsRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.GetRegisteredTaskExecutorsRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.GetTaskExecutorStatusRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.GetUnregisteredTaskExecutorsRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.InitializeTaskExecutorRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.ResourceOverviewRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.TaskExecutorAssignmentRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.TaskExecutorGatewayRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.TaskExecutorInfoRequest;
+import io.mantisrx.master.resourcecluster.ResourceClusterActor.TaskExecutorsList;
+import io.mantisrx.runtime.MachineDefinition;
+import io.mantisrx.server.core.domain.WorkerId;
+import io.mantisrx.server.master.resourcecluster.ClusterID;
+import io.mantisrx.server.master.resourcecluster.ResourceCluster;
+import io.mantisrx.server.master.resourcecluster.ResourceClusterTaskExecutorMapper;
+import io.mantisrx.server.master.resourcecluster.TaskExecutorID;
+import io.mantisrx.server.master.resourcecluster.TaskExecutorRegistration;
+import io.mantisrx.server.worker.TaskExecutorGateway;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+class ResourceClusterAkkaImpl extends ResourceClusterGatewayAkkaImpl implements ResourceCluster {
+
+  private final ClusterID clusterID;
+  private final ResourceClusterTaskExecutorMapper mapper;
+
+  public ResourceClusterAkkaImpl(
+      ActorRef resourceClusterManagerActor,
+      Duration askTimeout,
+      ClusterID clusterID,
+      ResourceClusterTaskExecutorMapper mapper) {
+    super(resourceClusterManagerActor, askTimeout, mapper);
+    this.clusterID = clusterID;
+      this.mapper = mapper;
+  }
+
+    @Override
+    public CompletableFuture<Ack> initializeTaskExecutor(TaskExecutorID taskExecutorID, WorkerId workerId) {
+        return Patterns.ask(
+                resourceClusterManagerActor,
+                new InitializeTaskExecutorRequest(taskExecutorID, workerId),
+                askTimeout)
+            .thenApply(Ack.class::cast)
+            .toCompletableFuture()
+            .whenComplete((ack, dontCare) ->
+                mapper.onTaskExecutorDiscovered(clusterID, taskExecutorID));
```

Comment:
What happens if there is an error from the future result?

Reply:
it doesn't matter. I still want the association to be made.